### ISSUE TITLE
Feature request: Basic auth support

### ISSUE THREAD

**Issue by silasbarnes** [securestate/king-phisher#27]
Hi,

I'd like to request an enhancement to include the option of enabling basic auth for visiting victims. The presentation of a native auth prompt has always enjoyed success during phishing campaigns and I think it would be very benficial. Basic auth over HTTPS would protect victim creds in transit while providing additional flexibility to the phisher.

Thanks,
Silas

**Comment by zeroSteiner** [comment]
I started looking into implementing this. It does not look like it's going to be a straight forward change so it might take a while.

**Comment by silasbarnes** [comment]
No dramas and no rush mate.

**Comment by zeroSteiner** [comment]
Alright so I'm happy to say I figured out how to implement this in a way that I think it will be easy and intuitive for users to use. I've documented the basic steps [in the wiki](https://github.com/securestate/king-phisher/wiki/Server-Pages-With-Jinja#requiring-basic-authentication). To summarize, a user just needs to add `{% set require_basic_auth = True %}` to their page and contents will only be displayed after the user has submitted creds which are logged.

This is currently in the dev branch and will be in the v0.3 release that I hope to have completed in the next week or two. Thanks for your patience and the feature suggestion!
